pkg/controller/statefulset/stateful_pod_control.go

```diff
@@ -194,6 +194,10 @@ func (spc *realStatefulPodControl) createPersistentVolumeClaims(set *apps.Statef
 		case err != nil:
 			errs = append(errs, fmt.Errorf("failed to retrieve PVC %s: %s", claim.Name, err))
 			spc.recordClaimEvent("create", set, pod, &claim, err)
+		case err == nil:
+			if pvc.DeletionTimestamp != nil {
+				errs = append(errs, fmt.Errorf("pvc %s is being deleted", claim.Name))
+			}
```

Comment:
```suggestion
		case err == nil:
			if pvc.DeletionTimestamp == nil {
				// consult apiserver to get the latest state
				pvc, err = spc.client.CoreV1().PersistentVolumeClaims(claim.Namespace).Get(context.TODO(), claim.Name, metav1.GetOptions{})
				if err != nil {
					errs = append(errs, fmt.Errorf("failed to retrieve PVC %s: %s", claim.Name, err))
					break
				}
			}
			if pvc.DeletionTimestamp != nil {
				errs = append(errs, fmt.Errorf("pvc %s is being deleted", claim.Name))
			}
```

Reply:
Why this is not added finally in the PR？

Reply:
Because this is not perfect too.  See https://github.com/kubernetes/kubernetes/pull/93457#discussion_r479962362